Library/Homebrew/formulary.rb

```diff
@@ -34,43 +34,64 @@ module Formulary
   # @api internal
   sig { void }
   def self.enable_factory_cache!
-    @factory_cache = true
+    @factory_cache_enabled = T.let(true, T.nilable(T::Boolean))
+    cache[platform_cache_tag] ||= {}
+    cache[platform_cache_tag][:formulary_factory] ||= {}
   end
 
+  sig { returns(T::Boolean) }
   def self.factory_cached?
-    !@factory_cache.nil?
+    !@factory_cache_enabled.nil?
   end
 
+  sig { returns(String) }
+  def self.platform_cache_tag
+    "#{Homebrew::SimulateSystem.current_os}_#{Homebrew::SimulateSystem.current_arch}"
+  end
+  private_class_method :platform_cache_tag
+
+  sig { returns(T::Hash[Symbol, T::Hash[T.any(String, Pathname), T.class_of(Formula)]]) }
```

Comment:
I agree that this Hash should probably be keyed on either String or Pathname rather than both.